doc/classes/Sprite2D.xml

```diff
@@ -15,12 +15,23 @@
 			</return>
 			<description>
 				Returns a [Rect2] representing the Sprite2D's boundary in local coordinates. Can be used to detect if the Sprite2D was clicked. Example:
-				[codeblock]
+				[codeblocks]
+				[gdscript]
 				func _input(event):
 				    if event is InputEventMouseButton and event.pressed and event.button_index == BUTTON_LEFT:
 				        if get_rect().has_point(to_local(event.position)):
 				            print("A click!")
-				[/codeblock]
+				[/gdscript]
+				[csharp]
+				if (@event is InputEventMouseButton &amp;&amp; (@event as InputEventMouseButton).Pressed &amp;&amp; (@event as InputEventMouseButton).ButtonIndex == (int)ButtonList.Left)
+				{
+				    if (GetRect().HasPoint(ToLocal((@event as InputEventMouseButton).Position)))
+				    {
+				        GD.Print("A click!");
```

Comment:
Too much casting going on here, the line is really long (164 chars), and there's no method boilerplate like in the C# example. Here's what I suggest instead:

```suggestion
				public override void _Input(InputEvent inputEvent)
				{
				    if (inputEvent is InputEventMouseButton inputEventMouse)
				    {
				        if (inputEventMouse.Pressed &amp;&amp; inputEventMouse.ButtonIndex == (int)ButtonList.Left)
				        {
				            if (GetRect().HasPoint(ToLocal(inputEventMouse.Position)))
				            {
				                GD.Print("A click!");
				            }
				        }
```